services/wireguard/endpoint/endpoint.go

```diff
@@ -39,13 +41,16 @@ type wgClient interface {
 }
 
 type connectionEndpoint struct {
-	iface             string
-	privateKey        string
-	publicIP          string
-	ipAddr            net.IPNet
-	endpoint          net.UDPAddr
-	resourceAllocator *resources.Allocator
-	wgClient          wgClient
+	iface              string
+	privateKey         string
+	publicIP           string
+	outboundIP         string
+	ipAddr             net.IPNet
+	endpoint           net.UDPAddr
+	resourceAllocator  *resources.Allocator
+	wgClient           wgClient
+	releasePortMapping func()
+	connectDelay       int // connect delay in milliseconds
```

Comment:
I know what problem are you trying to solve, but it will not be clear for another person who will try to read this code. We probably need to add some description of why we need this delay.